src/brogue/Items.c

```diff
@@ -5773,28 +5788,75 @@ void throwCommand(item *theItem) {
 
         confirmMessages();
 
-        thrownItem = generateItem(ALL_ITEMS, -1);
-        *thrownItem = *theItem; // clone the item
-        thrownItem->flags &= ~ITEM_EQUIPPED;
-        thrownItem->quantity = 1;
+        thrownItem = generateItem(ALL_ITEMS, -1);   // generate item object in memory
+        *thrownItem = *theItem;                     // clone the item
+        thrownItem->flags &= ~ITEM_EQUIPPED;        // item not equiped
+        thrownItem->quantity = 1;                   // item thrown, so quantity == 1
 
-        itemName(thrownItem, theName, false, false, NULL);
+        itemName(thrownItem, theName, false, false, NULL); // update name of the thrown item
 
         throwItem(thrownItem, &player, zapTarget, maxDistance);
     } else {
         return;
     }
 
-    // Now decrement or delete the thrown item out of the inventory.
+    // Update inventory
+    // -> Now decrement or delete the thrown item out of the inventory.
+    // -> Save last item thrown
     if (theItem->quantity > 1) {
         theItem->quantity--;
+        rogue.lastItemThrow = theItem; 
     } else {
+        rogue.lastItemThrow = NULL;
         removeItemFromChain(theItem, packItems);
         deleteItem(theItem);
     }
     playerTurnEnded();
 }
 
+void reThrowCommand(item *theItem) {
+    item *thrownItem;
+    unsigned char command[10];
+    char theName[COLS];
+    short maxDistance, zapTarget[2];
+
+    // Check last target
+    if (rogue.lastTarget == NULL)
+        return;
+
+    command[0] = RETHROW_KEY;
```

Comment:
By adding a new key here, this change is not backwards compatible. In theory we can record THROW_KEY here just fine.

Reply:
I think too, good idea.

Reply:
It's ok, recording and view recording is ok !